nautobot/extras/tests/test_views.py

```diff
@@ -988,28 +1036,234 @@ def setUpTestData(cls):
+            name="local/test_view/TestJob3",
             job_id=uuid.uuid4(),
             obj_type=obj_type,
         )
+
+
+class JobTestCase(
+    TestCase,
+):
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[])
+    def test_list_without_permission(self):
+        self.assertHttpStatus(self.client.get(reverse("extras:job_list")), 403)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_list(self):
+        response = self.client.get(reverse("extras:job_list"))
+        self.assertHttpStatus(response, 200)
+
+        response_body = extract_page_body(response.content.decode(response.charset))
+        self.assertIn("DummyJob", response_body)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[], JOBS_ROOT=DUMMY_JOBS)
+    def test_get_without_permission(self):
+        response = self.client.get(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+        self.assertHttpStatus(response, 403)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    def test_get(self):
+        response = self.client.get(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+        self.assertHttpStatus(response, 200)
+
+        response_body = extract_page_body(response.content.decode(response.charset))
+        self.assertIn("TestPass", response_body)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[], JOBS_ROOT=DUMMY_JOBS)
+    def test_post_without_permission(self):
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+        self.assertHttpStatus(response, 403)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    def test_run_without_schedule(self):
+        self.add_permissions("extras.run_job")
+
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+
+        self.assertHttpStatus(response, 200)
+        errors = extract_form_failures(response.content.decode(response.charset))
+        self.assertEqual(errors, ["_schedule_type: This field is required."])
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    def test_run_now_no_worker(self):
+        self.add_permissions("extras.run_job")
+
+        data = {
+            "_schedule_type": "immediately",
+        }
+
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}), data)
+
+        self.assertHttpStatus(response, 200)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    @mock.patch("nautobot.extras.utils.get_worker_count")
+    def test_run_now(self, patched):
+        self.add_permissions("extras.run_job")
+
+        data = {
+            "_schedule_type": "immediately",
+        }
+
+        patched.return_value = 1
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}), data)
+
```

Comment:
Maybe we should mock.patch JobResult.enqueue_job (or equivalent) and assert that it was called as expected?

Reply:
Patching `enqueue_job` would be a little problematic, since it’s responsible for creating a `JobResult`. But since the `JobResult` is created and we need to ensure that for the redirect, that might be just as good?

Reply:
I think it's possible to pass through from a patch to the underlying method to allow for normal operation (using `mock.DEFAULT`, IIRC) but I agree that just checking that the JobResult is created and the redirect is correct should suffice.

Reply:
Ah, thanks for the pointer! I’ll call it okay for now then :)